crates/windows-lib/src/switch/close.rs

```
@@ -54,3 +57,52 @@ pub fn close_switch(data: &mut WindowsSwitchData, switch: bool) {
         }
     }
 }
+
+pub fn close_switch_item(data: &WindowsSwitchData, event_sender: &Sender<TransferType>) {
+    if data.config.switch_workspaces {
+        kill_switch_workspace(data);
+    } else {
+        kill_switch_client(data);
+    }
+
+    let sender = event_sender.clone();
+    glib::timeout_add_local(Duration::from_millis(50), move || {
+        sender
+            .try_send(TransferType::RefreshSwitch(Box::new(
+                TransferType::CloseSwitchItem,
+            )))
+            .warn_details("Failed to send RefreshSwitch event");
+        glib::ControlFlow::Break
+    });
+}
+
+fn kill_switch_client(data: &WindowsSwitchData) {
+    if let Some(id) = data.active.client {
+        let addr = to_client_address(id);
+        let _ = kill_client(addr);
```

please propagate the error upwards and handle it at some point (using .warn_details("---"), which logs the error)